_finalproj/src/stage1.lua

```diff
@@ -40,37 +40,458 @@ function Stage1:init(def)
     
     self.stage:addEnemy(self.boss)
 
+    self.mech1counter = 1
```

Comment:
What are the purpose of these counters? Are they to keep track of phases for each AOE type?

They don't seem necessary. 

For example, it seems like you call `self:mech1()` once, and it just walks through the timers for the different phases

Reply:
without the counter, it will keep calling the current mechanic. basically, if my mechanic said spawn an aoe and then another 2 seconds later, it will keep infinitely spawning aoes and then infinitely more 2 seconds following. counters could be replaced as some boolean value but i think counters are a bit more flexible depending on whats getting implemented